### PR TITLE
Mef reader

### PR DESCRIPTION
**Pull request by jiecui**
Hi Robert @robertoostenveld,

Please try this pull request to test reading Mayo MEF dataset, which is updated to Fieldtrip version **1ba800674**:
1. Merge this pull request
    The external toolbox `mayo_mef` has been included in the PR, or can be downloaded from [GitHub repository of Multimodal Neuroimaging lab](https://github.com/MultimodalNeuroimagingLab/mef_reader_fieldtrip).
1. Download sample data from this [link](https://app.box.com/s/wwkdwl51joavrblu40v075bz2gb5fwrf)
    The data is hosted on box.com.  Please follow the download link above and look for 'Download' at the upper right corner of box.com page to download the data.  Unzip it if necessary.
1. Edit `test_mayo_mef.m` in folder `test/`
    In `test_mayo_mef.m` modify variables `mef21_data` and `mef30_data` to set the paths to the sample datasets.
1. Run `test_mayo_mef.m`
    The script should automatically compile the required C-MEX binary code.  Otherwise, please run `make_mayo_mex.m` at MATLAB command line.

More details may be also found on [GitHub repository of our lab](https://github.com/MultimodalNeuroimagingLab/mef_reader_fieldtrip).  Please let us know if you have any issue regarding this PR.

Many thanks,
Richard
--
MNL Group
Mayo Clinic Rochester USA
@dorahermes @MaxvandenBoom @jiecui

### CODE REVIEW

File: fileio/private/ft_hastoolbox.m [hunk 1]
```diff
@@ -166,10 +166,7 @@
   'EZC3D'         'see https://github.com/pyomeca/ezc3d'
   'GCMI'          'see https://github.com/robince/gcmi'
   'XSENS'         'see https://www.xsens.com/motion-capture and http://www.fieldtriptoolbox.org/getting_started/xsens/'
-  'MAYO_MEF'      'see https://github.com/MultimodalNeuroimagingLab/mef_reader_fieldtrip and https://msel.mayo.edu/codes.html'
-  'MATNWB'        'see https://neurodatawithoutborders.github.io/matnwb/'
-  'MATPLOTLIB'    'see https://nl.mathworks.com/matlabcentral/fileexchange/62729-matplotlib-perceptually-uniform-colormaps'
-  'CMOCEAN'       'see https://nl.mathworks.com/matlabcentral/fileexchange/57773-matplotlib-perceptually-uniform-colormaps'
```
Comment (robertoostenveld) [inline fileio/private/ft_hastoolbox.m:172]: why did you delete MATPLOTLIB and CMOCEAN?

**Comment by robertoostenveld** [comment]
You still seem to be having issues in properly using git and github: your Pull request not only adds your code, but also deletes code that we recently have been working on. That suggests that you are copying an old version of a file over the newer version in the repository. Please do not reinsert old copies of files, make the changes to the latest versions that are in the master branch. 

Can you please make a new pull request, with only a *single* functional change to a single file: please take ft_preprocessing and add the cfg.password option, and add it to the calls to ft_read_header and ft_read_data?

That change shoudl be limited to about 5 lines in the code of ft_preprocessing. If you submit that, I will review it.

**Comment by jiecui** [comment]
Dear Robert,
I'm sorry for the confusion.  I think there are some mistakes in my Github operations.  We will review it first in our lab and make sure our suggestion will accommodate the newest version of Fieldtrip.  After that, we will make a new request.

Thanks for taking time to work on our request.

Regards,
Richard
@dorahermes @MaxvandenBoom
